src/main/java/org/sonar/plugins/stash/issue/MarkdownPrinter.java

```diff
@@ -119,20 +131,30 @@ public static String printReportMarkdown(PullRequestRef pr, String stashURL, Str
       sb.append("| Issues list |").append(NEW_LINE);
       sb.append("|------------|").append(NEW_LINE);
       for (String severity: orderedSeverities) {
-        sb.append(printIssueListBySeverityMarkdown(generalIssues, sonarQubeURL, severity));
+        int maxLength = SOFT_SUMMARY_COMMENT_MAX_LENGTH / 2 - sb.length();

```

Comment:
I would prefer to have the truncation/maxlength logic in its own wrapper class around StringBuffer.
With the signature:
```java
class SizeSensitiveStringBuffer() {
    public SizeSensitiveStringBuffer(int maxLength);
    public boolean append(String content, int maxLength, String alternativeContent);
    public void append(String content);
}
```